sdk/src/main/scala/com/horizen/AbstractSidechainNodeViewHolder.scala

```diff
@@ -137,6 +140,17 @@ abstract class AbstractSidechainNodeViewHolder[
       }
   }
 
+  protected def processGenerateSecret: Receive = {
+    case AbstractSidechainNodeViewHolder.ReceivableMessages.GenerateSecret(secretCreator) =>
+      vault().generateNextSecret(secretCreator) match {
+        case Success(newVault) =>
```

Comment:
We should get bot the wallet and the generated secret from the success case, and relay a Success(secrte) to the sender, not a Unit.

Reply:
Done